File: frontend/src/auth/roles.ts
// Mirrors GrupoJuridico.Crm.Domain.Constants.Roles on the backend — keep in sync.
export const Roles = {
  Admin: 'Admin',
  Supervisor: 'Supervisor',
  Editor: 'Editor',
  Asesor: 'Asesor',
} as const;

export type Role = (typeof Roles)[keyof typeof Roles];

export function isAdmin(role: Role): boolean {
  return role === Roles.Admin;
}

export function isSupervisor(role: Role): boolean {
  return role === Roles.Supervisor;
}

export function isEditor(role: Role): boolean {
  return role === Roles.Editor;
}

export function isAsesor(role: Role): boolean {
  return role === Roles.Asesor;
}

// Admin y Supervisor pueden filtrar el tablero/panel por asesor, ver el equipo, etc.
export function canFilterTeam(role: Role): boolean {
  return isAdmin(role) || isSupervisor(role);
}

export function roleLabel(role: Role): string {
  switch (role) {
    case Roles.Admin:
      return 'Administrador';
    case Roles.Supervisor:
      return 'Supervisor';
    case Roles.Editor:
      return 'Editor';
    case Roles.Asesor:
      return 'Asesor';
  }
}

File: backend/src/GrupoJuridico.Crm.Domain/Constants/Roles.cs
namespace GrupoJuridico.Crm.Domain.Constants;

// Admin: acceso total. Supervisor: gestiona a sus asesores. Editor: solo registra clientes.
// Asesor: dueño de sus propios clientes en el tablero.
public static class Roles
{
    public const string Admin = "Admin";
    public const string Supervisor = "Supervisor";
    public const string Editor = "Editor";
    public const string Asesor = "Asesor";
}
